scan-engine/src/main/java/de/tillhub/scanengine/default/ui/GoogleScanningViewModel.kt

```diff
@@ -45,21 +45,19 @@ private class QRImageAnalyzer(
         if (mediaImage != null) {
             val image = InputImage.fromMediaImage(mediaImage, imageProxy.imageInfo.rotationDegrees)
 
-            scanner.process(image)
-                .addOnSuccessListener { list ->
-                    if (list.size > 0 && list[0].rawValue != null) {
-                        scanningState.value = ScanningState.CodeScanned(list[0].rawValue!!)
-                        scanner.close()
-                    }
-                }
-                .addOnCompleteListener {
-                    imageProxy.close()
+            scanner.process(image).addOnSuccessListener { list ->
+                if (list.size > 0 && list[0].rawValue != null) {
+                    scanningState.value = ScanningState.CodeScanned(list[0].rawValue!!)
+                    scanner.close()
                 }
+            }.addOnCompleteListener {
+                imageProxy.close()
+            }
         }
     }
 }
 
-sealed class ScanningState {
-    object Idle : ScanningState()
-    data class CodeScanned(val barcode: String) : ScanningState()
+internal sealed class ScanningState {
+    data object Idle : ScanningState()
+    class CodeScanned(val barcode: String) : ScanningState()
```

Comment:
This is ok to be data class if it's used internal. Only classes that are exposed outside of the sdk should not be data class